### PR TITLE
Add validation for current-or-past dates

### PR DESCRIPTION
**Pull request by raquelromano-gov**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1233

This adds a validator and unit test,and updates 1 field in the UI (see below screenshot)

Still todo: use the current-or-past validator for the service period date range, but will open a separate PR for that.

Note: I need to commit some time to getting e2e tests running locally, so I won't merge this until that's done and they're passing.

Screenshots:
image
image

### CODE REVIEW

**Review by jbalboni — COMMENTED** [review]
Looks good, had a couple questions.

File: src/js/edu-benefits/1995/config/form.js [hunk 1]
```diff
@@ -258,7 +259,9 @@ const formConfig = {
               },
               address: address.uiSchema()
             },
-            trainingEndDate: date.uiSchema('When did you stop taking classes or participating in the training program?'),
+            trainingEndDate: _.merge(date.uiSchema('When did you stop taking classes or participating in the training program?'), {
```
Comment (raquelromano-gov) [inline src/js/edu-benefits/1995/config/form.js]: Yeah, sounds like a good idea. Will do that.

File: test/common/utils/validations.unit.spec.js [hunk 2]
```diff
@@ -195,16 +196,16 @@ describe('Validations unit tests', () => {
     it('should validate partial range', () => {
       const fromDate = {
         month: {
-          value: '2'
+          value: 2
```
Comment (raquelromano-gov) [inline test/common/utils/validations.unit.spec.js]: Just for consistency as there seemed to be places they were strings and places they were numbers. Perhaps they should all be sent as strings then?

Comment (raquelromano-gov) [inline test/common/utils/validations.unit.spec.js]: @jbalboni LMK if you think these should be strings or numbers.

File: src/js/edu-benefits/1995/config/form.js (hunk 1 above)
Comment (raquelromano-gov) [inline src/js/edu-benefits/1995/config/form.js]: Done

File: test/common/utils/validations.unit.spec.js (hunk 2 above)
Comment (raquelromano-gov) [inline test/common/utils/validations.unit.spec.js]: PTAL - this makes the test kind of messy (esp. the places where values generated by moment() are converted to strings), but at least it's consistent now and emulates the components more accurately.

File: src/js/common/schemaform/validation.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { Validator } from 'jsonschema';
 
 import { retrieveSchema } from 'react-jsonschema-form/lib/utils';
 
-import { isValidSSN, isValidPartialDate, isValidDateRange, isValidRoutingNumber, isValidUSZipCode, isValidCanPostalCode } from '../utils/validations';
+import { isValidSSN, isValidPartialDate, isValidCurrentOrPastDate, isValidDateRange, isValidRoutingNumber, isValidUSZipCode, isValidCanPostalCode } from '../utils/validations';
```
Comment (jbalboni) [inline src/js/common/schemaform/validation.js]: We should probably reformat this so each function is on its own line.

**Review by jbalboni — APPROVED** [review]
LGTM.

File: src/js/common/schemaform/validation.js (hunk 3 above)
Comment (raquelromano-gov) [inline src/js/common/schemaform/validation.js]: Done

**Pull request merged**